pytest_helm_charts/giantswarm_app_platform/utils.py

```diff
@@ -1,30 +1,192 @@
 ) -> List[AppCR]:
+    """
+    Block until all the apps are running or timeout is reached.
+
+    Args:
+        kube_client: client to use to connect to the k8s cluster
+        app_names: a list of application names to check; all of the applications must be running for this
+        function to succeed
+        app_namespace: namespace where the App CRs of all the apps are stored
+        timeout_sec: timeout for the call
+        missing_ok: when `True`, the function ignores that some of the apps listed in the `app_names`
+        don't exist in k8s API and waits for them to show up; when `False`, an
+        [ObjectNotFound](pykube.exceptions.ObjectDoesNotExist) exception is raised.
+
+    Returns:
+        The list of App CRs with all the apps listed in `app_names` included.
+
+    Raises:
+        TimeoutError: when timeout is reached.
+        pykube.exceptions.ObjectDoesNotExist: when `missing_ok == False` and one of the apps
+        listed in `app_names` can't be found in k8s API
+
+    """
     apps = wait_for_namespaced_objects_condition(
         kube_client, AppCR, app_names, app_namespace, _app_deployed, timeout_sec, missing_ok
     )
     return apps
+
+
+def wait_for_app_to_be_deleted(
+    kube_client: HTTPClient,
+    app_name: str,
+    app_namespace: str,
+    timeout_sec: int,
+) -> bool:
+    """
+    Block until an App CR has status `deleted` or doesn't exist in k8s API.
+
+    Args:
+        kube_client: client to use to connect to the k8s cluster
+        app_name: an application name to check
+        app_namespace: namespace where the App CRs of all the apps are stored
+        timeout_sec: timeout for the call
+
+    Returns:
+        `True` when the App CR was found with status `deleted` or was not found at all. `False` otherwise.
+
+    Raises:
+        TimeoutError: when timeout is reached.
+
+    """
+    try:
+        apps = wait_for_namespaced_objects_condition(
+            kube_client, AppCR, [app_name], app_namespace, _app_deleted, timeout_sec, missing_ok=False
+        )
+    except pykube.exceptions.ObjectDoesNotExist:
+        return True
+    return len(apps) == 1
+
+
+def delete_app(configured_app: ConfiguredApp) -> None:
+    """
+    Deletes the app created by [create_app](create_app).
+    Args:
+        configured_app: ConfiguredApp (with optional ConfigMap configuration) to be deleted.
+
+    Returns:
+        None
+    """
+    configured_app.app.delete()
+    if configured_app.app_cm:
+        configured_app.app_cm.delete()
+    # TODO: wait until finalizer is gone
+
+
+def create_app(
+    kube_client: HTTPClient,
+    app_name: str,
+    app_version: str,
+    catalog_name,
+    namespace: str,
+    deployment_namespace: str,
+    config_values: YamlDict = None,
+    namespace_config_annotations: YamlDict = None,
+    namespace_config_labels: YamlDict = None,
+) -> ConfiguredApp:
+    """Creates and deploys a new app using App CR. Optionally creates configuration ConfigMap. Calls are blocking.
```

Comment:
```suggestion
    """Creates and deploys a new app using App CR. Optionally creates a values ConfigMap. Calls are blocking.
```

In docs we usually call this the values configmap.